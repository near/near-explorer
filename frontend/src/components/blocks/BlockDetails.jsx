import moment from "moment";
import { Row, Col } from "react-bootstrap";

import BlockCard from "./BlockCard";

export default ({ block }) => {
  return (
    <>
      <Row noGutters="true">
        <Col className="block-info-container">
          <Row noGutters="true">
            <Col md="3">
              <BlockCard
                title="Transactions"
                imgLink="/static/images/icon-m-transaction.svg"
                text={block.transactionsCount}
                cls="block-card-no-side-border block-card-no-top-border"
              />
            </Col>
            <Col md="2">
              <BlockCard
                title="Size"
                imgLink="/static/images/icon-m-size.svg"
                text="TODO"
                cls="block-card-no-top-border"
              />
            </Col>
          </Row>
          <Row noGutters="true">
            <Col md="4">
              <BlockCard
                title="Created"
                text={moment(block.timestamp).format(
                  "MMMM DD, YYYY [at] h:mm:ssa"
                )}
                textCls="block-card-created-text"
                cls="block-card-no-border"
              />
            </Col>
            <Col md="8">
              <BlockCard
                title="Hash"
                text={block.hash}
                textCls="block-card-created-text"
                cls="block-card-no-border"
              />
            </Col>
          </Row>
          <Row noGutters="true">
            <Col md="12">
              <BlockCard
                title="Parent Hash"
                text={block.prevHash}
                textCls="block-card-parent-hash-text"
                textLink={true}
                cls="block-card-parent-hash block-card-no-border"
              />
            </Col>
          </Row>
        </Col>
      </Row>
      <style jsx global>{`
        .block-card-created-text {
          font-size: 18px;
          font-weight: 500;
          color: #4a4f54;
        }

        .block-card-author-text {
          color: #0072ce;
        }

        .block-card-parent-hash-text {
          font-size: 18px;
          font-weight: 500;
          color: #6ad1e3;
          cursor: pointer;
          text-decoration: none !important;
        }

        .block-card-parent-hash-text:hover {
          color: #6ad1e3;
        }

        .block-card-parent-hash {
          background-color: #f8f8f8;
        }

        .block-card-no-border {
          border: 0 !important;
        }

        .block-card-no-side-border {
          border-left: 0 !important;
          border-right: 0 !important;
        }

        .block-card-no-top-border {
          border-top: 0 !important;
        }

        .block-card-no-bottom-border {
          border-bottom: 0 !important;
        }

        .block-info-container {
          border: solid 4px #e6e6e6;
          border-radius: 4px;
        }
      `}</style>
    </>
  );
};
