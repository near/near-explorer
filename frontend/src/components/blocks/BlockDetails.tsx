import moment from "moment";

import { Row, Col } from "react-bootstrap";

import * as B from "../../libraries/explorer-wamp/blocks";

import BlockLink from "../utils/BlockLink";
import CardCell from "../utils/CardCell";
import Term from "../utils/Term";

export interface Props {
  block: B.BlockInfo;
}

export default ({ block }: Props) => {
  return (
    <>
      <Row noGutters>
        <Col className="block-info-container">
          <Row noGutters className="block-info-header">
            <Col md="4">
              <CardCell
                title={
                  <Term title={"Transactions"}>
                    {"Number of transactions in this block. "}
                    <a
                      href={
                        "https://docs.nearprotocol.com/docs/concepts/transaction"
                      }
                    >
                      docs
                    </a>
                  </Term>
                }
                imgLink="/static/images/icon-m-transaction.svg"
                text={block.transactionsCount.toLocaleString()}
                className="border-0"
              />
            </Col>
            <Col md="4">
              <CardCell
                title={
                  <Term title={"Gas Used"}>
                    {"Total units of gas used by transactions in this block. "}
                    <a href={"https://docs.nearprotocol.com/docs/concepts/gas"}>
                      docs
                    </a>
                  </Term>
                }
                imgLink="/static/images/icon-m-size.svg"
                text={block.gasUsed.toLocaleString()}
              />
            </Col>
            <Col md="4">
              <CardCell
                title={
                  <Term title={"Gas Price"}>
                    {"Cost per unit of gas. "}
                    <a href={"https://docs.nearprotocol.com/docs/concepts/gas"}>
                      docs
                    </a>
                  </Term>
                }
                imgLink="/static/images/icon-m-filter.svg"
                text={block.gasPrice.toLocaleString()}
              />
            </Col>
          </Row>
          <Row noGutters className="border-0">
            <Col md="4">
              <CardCell
                title={
                  <Term title={"Created"}>
                    {"Timestamp of when this block finalized. "}
                  </Term>
                }
                text={moment(block.timestamp).format(
                  "MMMM DD, YYYY [at] h:mm:ssa"
                )}
                className="block-card-created border-0"
              />
            </Col>
            <Col md="8">
              <CardCell
                title={
                  <Term title={"Hash"}>
                    {"Unique identifier (hash) of this block. "}
                  </Term>
                }
                text={block.hash}
                className="border-0"
              />
            </Col>
          </Row>
          <Row noGutters>
            <Col md="12">
              <CardCell
                title={
                  <Term title={"Parent Hash"}>
                    {"Unique identifier (hash) of previous block. "}
                  </Term>
                }
                text={
                  block.prevHash === "11111111111111111111111111111111" ? (
                    "Genesis"
                  ) : (
                    <BlockLink blockHash={block.prevHash}>
                      {block.prevHash}
                    </BlockLink>
                  )
                }
                className="block-card-parent-hash border-0"
              />
            </Col>
          </Row>
        </Col>
      </Row>
      <style jsx global>{`
        .block-info-container {
          border: solid 4px #e6e6e6;
          border-radius: 4px;
        }

        .block-info-container > .row {
          border-bottom: 2px solid #e6e6e6;
        }

        .block-info-container > .row:last-of-type {
          border-bottom: 0;
        }

        .block-info-header .card-cell-text {
          font-size: 24px;
        }

        .block-card-created-text {
          font-size: 18px;
          font-weight: 500;
          color: #4a4f54;
        }

        .block-card-parent-hash {
          background-color: #f8f8f8;
        }
      `}</style>
    </>
  );
};
