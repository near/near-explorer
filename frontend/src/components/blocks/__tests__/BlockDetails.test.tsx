import renderer from "react-test-renderer";

import BlockDetails from "../BlockDetails";

describe("<BlockDetails />", () => {
  it("renders", () => {
    expect(
      renderer.create(
        <BlockDetails
          block={{
            transactionsCount: 3,
            timestamp: Number(new Date(2019, 1, 1)),
            hash: "EVvWW1S9BFaEjY1JBNSdstb7ZTtTFjQ6cygkbw1KY4tL",
            prevHash: "EVvWW1S9BFaEjY1JBNSdstb7Zjghjlyguiygfhgu",
            height: 12345
          }}
        />
      )
    ).toMatchSnapshot();
  });
});
