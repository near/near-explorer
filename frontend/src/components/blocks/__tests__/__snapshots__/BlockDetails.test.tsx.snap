// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlockDetails /> renders 1`] = `
<div
  className="row no-gutters"
>
  <div
    className="block-info-container col"
  >
    <div
      className="block-info-header row no-gutters"
    >
      <div
        className="col-md-3"
      >
        <div
          className="card-cell border-0 card"
        >
          <div
            className="card-body"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="card-cell-title align-self-center col-md-12 col-auto"
              >
                <img
                  className="jsx-1276427197 card-cell-title-img"
                  src="/static/images/icon-m-transaction.svg"
                />
                Transactions
                <div
                  className="jsx-782042476 term-helper"
                >
                  <img
                    className="jsx-782042476 info"
                    onClick={[Function]}
                    src="/static/images/icon-info.svg"
                  />
                </div>
              </div>
              <div
                className="ml-auto card-cell-text align-self-center col-md-12 col-auto"
              >
                3
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-md-3"
      >
        <div
          className="card-cell  card"
        >
          <div
            className="card-body"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="card-cell-title align-self-center col-md-12 col-auto"
              >
                <img
                  className="jsx-1276427197 card-cell-title-img"
                  src="/static/images/icon-m-size.svg"
                />
                Gas Used
                <div
                  className="jsx-782042476 term-helper"
                >
                  <img
                    className="jsx-782042476 info"
                    onClick={[Function]}
                    src="/static/images/icon-info.svg"
                  />
                </div>
              </div>
              <div
                className="ml-auto card-cell-text align-self-center col-md-12 col-auto"
              >
                0
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-md-3"
      >
        <div
          className="card-cell  card"
        >
          <div
            className="card-body"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="card-cell-title align-self-center col-md-12 col-auto"
              >
                <img
                  className="jsx-1276427197 card-cell-title-img"
                  src="/static/images/icon-m-filter.svg"
                />
                Gas Price
                <div
                  className="jsx-782042476 term-helper"
                >
                  <img
                    className="jsx-782042476 info"
                    onClick={[Function]}
                    src="/static/images/icon-info.svg"
                  />
                </div>
              </div>
              <div
                className="ml-auto card-cell-text align-self-center col-md-12 col-auto"
              >
                5000
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-md-3"
      >
        <div
          className="card-cell  card"
        >
          <div
            className="card-body"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="card-cell-title align-self-center col-md-12 col-auto"
              >
                <img
                  className="jsx-1276427197 card-cell-title-img"
                  src="/static/images/icon-t-status.svg"
                />
                Status
                <div
                  className="jsx-782042476 term-helper"
                >
                  <img
                    className="jsx-782042476 info"
                    onClick={[Function]}
                    src="/static/images/icon-info.svg"
                  />
                </div>
              </div>
              <div
                className="ml-auto card-cell-text align-self-center col-md-12 col-auto"
              >
                Finalized
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="border-0 row no-gutters"
    >
      <div
        className="col-md-4"
      >
        <div
          className="card-cell block-card-created border-0 card"
        >
          <div
            className="card-body"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="card-cell-title align-self-center col-md-12 col-auto"
              >
                Created
                <div
                  className="jsx-782042476 term-helper"
                >
                  <img
                    className="jsx-782042476 info"
                    onClick={[Function]}
                    src="/static/images/icon-info.svg"
                  />
                </div>
              </div>
              <div
                className="ml-auto card-cell-text align-self-center col-md-12 col-auto"
              >
                February 01, 2019 at 12:00:00am
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-md-8"
      >
        <div
          className="card-cell border-0 card"
        >
          <div
            className="card-body"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="card-cell-title align-self-center col-md-12 col-auto"
              >
                Hash
                <div
                  className="jsx-782042476 term-helper"
                >
                  <img
                    className="jsx-782042476 info"
                    onClick={[Function]}
                    src="/static/images/icon-info.svg"
                  />
                </div>
              </div>
              <div
                className="ml-auto card-cell-text align-self-center col-md-12 col-auto"
              >
                EVvWW1S9BFaEjY1JBNSdstb7ZTtTFjQ6cygkbw1KY4tL
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row no-gutters"
    >
      <div
        className="col-md-12"
      >
        <div
          className="card-cell block-card-parent-hash border-0 card"
        >
          <div
            className="card-body"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="card-cell-title align-self-center col-md-12 col-auto"
              >
                Parent Hash
                <div
                  className="jsx-782042476 term-helper"
                >
                  <img
                    className="jsx-782042476 info"
                    onClick={[Function]}
                    src="/static/images/icon-info.svg"
                  />
                </div>
              </div>
              <div
                className="ml-auto card-cell-text align-self-center col-md-12 col-auto"
              >
                <a
                  className="block-link"
                  href="/blocks/EVvWW1S9BFaEjY1JBNSdstb7Zjghjlyguiygfhgu"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  EVvWW1S9BFaEjY1JBNSdstb7Zjghjlyguiygfhgu
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
